nautobot/dcim/models/devices.py

```diff
@@ -979,56 +981,56 @@ def all_console_ports(self):
         Return all Console Ports that are installed in the device or in modules that are installed in the device.
         """
         # TODO: These could probably be optimized to reduce the number of joins
-        return ConsolePort.objects.filter(models.Q(device=self) | models.Q(module__in=self.all_modules))
+        return ConsolePort.objects.filter(Q(device=self) | Q(module__in=self.all_modules))
 
     @property
     def all_console_server_ports(self):
         """
         Return all Console Server Ports that are installed in the device or in modules that are installed in the device.
         """
-        return ConsoleServerPort.objects.filter(models.Q(device=self) | models.Q(module__in=self.all_modules))
+        return ConsoleServerPort.objects.filter(Q(device=self) | Q(module__in=self.all_modules))
 
     @property
     def all_front_ports(self):
         """
         Return all Front Ports that are installed in the device or in modules that are installed in the device.
         """
-        return FrontPort.objects.filter(models.Q(device=self) | models.Q(module__in=self.all_modules))
+        return FrontPort.objects.filter(Q(device=self) | Q(module__in=self.all_modules))
 
     @property
     def all_interfaces(self):
         """
         Return all Interfaces that are installed in the device or in modules that are installed in the device.
         """
-        return self.vc_interfaces | Interface.objects.filter(module__in=self.all_modules)
+        return Interface.objects.filter(Q(device=self) | Q(module__in=self.all_modules))
```

Comment:
Behavior change here. Since we already had the behavior `'interfaces' are installed in this device. 'vc_interfaces' are installed in any member of the virtual chassis` I made some changes to be consistent with that behavior but adding modular component support.